flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/local/result/MaterializedCollectStreamResult.java

```diff
@@ -224,7 +224,7 @@ private void processDelete(Row row) {
 		}
 
 		for (int i = startSearchPos; i >= validRowPosition; i--) {
-			if (materializedTable.get(i).equals(row)) {
+			if (materializedTable.get(i).fieldsEquals(row)) {
 				materializedTable.remove(i);
 				rowPositionCache.remove(row);
```

Comment:
@zjffdu I looked into this topic again in depth. I think your fix is not enough. Because `rowPositionCache.remove(row);` needs to ignore the row kind. I wonder if we can fix the problem in a different layer. @godfreyhe how about we erase the row kind for `Tuple2<Boolean, Row>`? In the future, we should have a `tableEnv.toDataStream(ChangelogMode): Row` so the `Tuple2` case will not be needed. Until then we should not encode the deletion twice in the tuple and in the row. What do you think?

Reply:
@twalthr for now, we can erase the row kind for `Tuple2<Boolean, Row>`. After https://github.com/apache/flink/pull/12199 merged, I want to refactor the select logic in sql client. Then `Tuple2` is not needed, we can use `Row` for all cases.

Reply:
@godfreyhe Is #12199 going to be merged in 1.11 ? For me, this is an api chagne, it would be better to stablize it.

Reply:
@zjffdu , I want to merge https://github.com/apache/flink/pull/12199 into 1.11, but 1.11 is in  feature freeze since last week. This pr will only be merged into master.

Reply:
Then we should just erase the changeflag in `Tuple2<Boolean, Row>` as fix for 1.11. @zjffdu  can you update this PR?

Reply:
@twalthr Do you mean to change `Tuple2<Boolean, Row>` to `Row` ?

Reply:
No, just remove the flag that is encoded in `Row` and purely depend on the `Boolean` of `Tuple2`.

Reply:
@twalthr  Is it safe to do that ? I see many places use this field.
![image](https://user-images.githubusercontent.com/164491/84008773-e2060f00-a9a4-11ea-8f10-0b239a183d49.png)

Reply:
This field is valid in most locations. Just not for `Tuple2<Boolean, Row>` generation, these path should be touched. Not the other non-legacy paths. Where is `Tuple2<>` created, there is some MapFunction in the code base doing this.

Reply:
Sorry @twalthr I don't understand, if this fields is valid and necessary, then how can I remove this field in `Row`

Reply:
Sorry, maybe I was not specific enough. I didn't mean to remove the field but I meant reseting the value of `RowKind`. All rows returned by `Tuple2<Boolean, Row>` should have `Row.ofKind(RowKind.INSERT, ...)`. And users should rely on the boolean flag instead. I hope this makes it clearer? otherwise I can also assign this ticket to myself to speed things up?

Reply:
@twalthr Thanks for the clarification, I have updated the PR.

Reply:
ping @twalthr